### ISSUE TITLE
Inline templates with string interpolations can result in incorrect fixes

### ISSUE THREAD

**Issue by reduckted** [angular-eslint/angular-eslint#2325]
**Description and reproduction of the issue**

Consider this code:
```ts
import { Component } from '@angular/core';
import { describe, it } from 'node:test';

describe('test', () => {
  ['alpha', 'beta'].forEach((name) => {
    it('is a dynamic test', () => {
      @Component({
        template: `<app-component class="${name}"></app-component>`,
      })
      class TestComponent {}
    });
  });
});
```

The component's template is defined using a template literal with interpolations. When this component's template is turned into a template that can be parsed by ESLint, the template is seen as either:
```html
<app-component class="alpha"></app-component>
```
or
```html
<app-component class="beta"></app-component>
```

Any problems that are reported by the rules will use spans that come from the parsed template, but any fixes will be applied to the original template literal. Because the parsed template is a different length to the original template literal, this can result in fixes changing the wrong code. 

For example, using the code above with the `prefer-self-closing-tags` rule, the fix changes the template to:
```html
<app-component class="${name}"> />
```

Reproduction is here:
https://stackblitz.com/edit/angular-eslint-repros-snkd99lv?file=src%2Fmain.test.ts

Simply open the terminal and run `ng lint --fix`.

**Versions**

| package                                  | version |
| ---------------------------------------- | ------- |
| `@angular-eslint/eslint-plugin-template` | `19.2.1` |
| `@angular-eslint/template-parser`        | `19.2.1` |
| `@typescript-eslint/parser`              | `8.16.0` |
| `ESLint`                                 | `9.15.0` |
| `node`                                   | `18.20.3` |

```sh
Angular CLI: 19.0.0
Node: 18.20.3
Package Manager: npm 10.2.3
OS: linux x64

Angular: 19.0.0
... animations, build, cli, common, compiler, compiler-cli, core
... forms, platform-browser, router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1900.0
@angular-devkit/core         19.0.0
@angular-devkit/schematics   19.0.0
@schematics/angular          19.0.0
rxjs                         7.8.1
typescript                   5.6.3
zone.js                      0.15.0

```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by reduckted** [comment]
I don't think it's possible to adjust the fixes to account for the interpolations (in fact, it's probably far too complicated to even consider as an option). One solution might be to remove the fixes in the post-processor if the original string is a template literal with interpolations.

The downside is that you would have to manually fix the problems, but the upside is that your code won't break. 😁

**Comment by reduckted** [comment]
Well I was completely wrong on what would need to change for this. Turns out it was caused by the inline template extraction including the opening and closing backticks. A very simple fix in #2368. Phew!
